lib/kafka/gzip_codec.rb

```diff
@@ -0,0 +1,28 @@
+module Kafka
+  class GzipCodec
```

Comment:
it's not about being confusing but about providing a nice and clean way to extend this gem. What if I would like to implement my own compressing layer? I would just inherit Kafka::Codecs::Base, implement what I need and that should be all...